include/unix/osd.h

```diff
@@ -304,7 +306,46 @@ ofi_cpuid(unsigned func, unsigned subfunc, unsigned cpuinfo[4])
 	asm volatile("clflush %0" : "+m" (*(volatile char *) addr))
 #define ofi_sfence() asm volatile("sfence" ::: "memory")
 
-#else /* defined(__x86_64__) || defined(__amd64__) */
+#elif ( defined(__riscv) && defined(__riscv_xlen) && (__riscv_xlen == 64) && (defined(__linux__) || defined(__APPLE__)) )
+
+#include <stdio.h>
+#include <regex.h>
+
+static inline void
+ofi_cpuid(unsigned func, unsigned subfunc, unsigned cpuinfo[4])
+{
+     FILE *f = fopen("/proc/cpuinfo", "r");
+
+     regex_t r;
+     regmatch_t  match[1];
+
+     regcomp(&r, "hart", 0);
+     unsigned int hart = 0;
```

Comment:
please relocate all variable declarations to the top of the function to make it faster/easier to locate.